pyccl/base.py

```diff
@@ -43,3 +46,254 @@ def hash_(obj):
     """Generic hash method, which changes between processes."""
     digest = hash(repr(_to_hashable(obj))) + sys.maxsize + 1
     return digest
+
+
+class _ClassPropertyMeta(type):
+    """Implement `property` to a `classmethod`."""
+    # TODO: in py39+ decorators `classmethod` and `property` can be combined
```

Comment:
can you make it more explicit that we want to drop this by the time we make 3.9 the minimum requirement?

Reply:
Looks like we're going to have to stick with it, the Python devs removed the ability to wrap `property` inside `classmethod` in Python 3.11.
https://docs.python.org/3.11/library/functions.html#classmethod
https://github.com/python/cpython/issues/89519

Reply:
Update: Yes, so it looks like after Py3.11, metaclasses are the only reliable way to combine `classmethod` and `property`.

Reply:
hahaha